src/segmenter/utils.rs

```diff
@@ -0,0 +1,73 @@
+use fst::raw::{Fst, Output};
+
+/// Final-state-transducer (FST) Segmenter
+pub(crate) struct FstSegmenter<'fst> {
+    words_fst: &'fst Fst<&'fst [u8]>
+}
+
+impl<'fst> FstSegmenter<'fst> {
+    pub(crate) fn new(words_fst: &'fst Fst<&'fst [u8]>) -> Self {
+        Self { words_fst }
+    }
+}
+
+impl<'fst> FstSegmenter<'fst> {
```

Comment:
A small last change, let's merge these two `impl` 
```suggestion
impl<'fst> FstSegmenter<'fst> {
    pub(crate) fn new(words_fst: &'fst Fst<&'fst [u8]>) -> Self {
        Self { words_fst }
    }

```

Reply:
Oh right, forgot there's no need to have multiple impl blocks now that there's no anonymous lifetime, I've unified the impl blocks, thanks!